File: app/src/main/java/com/example/snapchatclone/MainActivity.java
package com.example.snapchatclone;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.viewpager2.widget.ViewPager2;
import androidx.fragment.app.FragmentTransaction;
import android.app.Activity;

import android.content.Intent;
import android.os.Bundle;

import com.google.android.material.tabs.TabLayout;
import com.google.android.material.tabs.TabLayoutMediator;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        //crate and get elements

        //TabLayout tabLayout=findViewById(R.id.tabs);
        //add to main activity xml if I want add tabs
        /** <com.google.android.material.tabs.TabLayout
         android:id="@+id/tabs"
         android:layout_width="match_parent"
         android:layout_height="wrap_content">

         </com.google.android.material.tabs.TabLayout>
         also add android:layout_below="@+id/tabs" to viewpager2 **/

        ViewPager2 viewPager2;
        viewPager2= findViewById(R.id.viewPager2);
        //before need create a layout for games and leading board
        ViewPageAdapter adapter= new ViewPageAdapter(this);
        viewPager2.setAdapter(adapter);
        //below make tabs move tab1,tab2,tab3
        /** new TabLayoutMediator(tabLayout, viewPager2, new TabLayoutMediator.TabConfigurationStrategy() {
            @Override
            public void onConfigureTab(@NonNull TabLayout.Tab tab, int position) {
                tab.setText("Tab"+(position +1));
            }
        }).attach(); **/

    }

}

File: app/src/main/java/com/example/snapchatclone/MainActivity.kt
package com.example.memorygame
import android.os.*
import com.google.android.material.snackbar.Snackbar
import androidx.appcompat.app.AppCompatActivity
import androidx.navigation.findNavController
import androidx.navigation.ui.AppBarConfiguration
import androidx.navigation.ui.navigateUp
import androidx.navigation.ui.setupActionBarWithNavController
import android.view.Menu
import android.view.MenuItem
import android.widget.TextView
import com.example.memorygame.databinding.ActivityMainBinding
import com.example.memorygame.R.drawable.*
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.activity_main.view.*
import kotlinx.android.synthetic.main.content_main.*
import java.security.KeyStore

class MainActivity : AppCompatActivity() {

    private lateinit var appBarConfiguration: AppBarConfiguration
    private lateinit var binding: ActivityMainBinding
    internal var gameStarted=false
    lateinit var textView : TextView
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        textView = findViewById(R.id.timer)
        object : CountDownTimer(30000, 1000) {
            // Callback function, fired on regular interval
            override fun onTick(millisUntilFinished: Long) {
                textView.setText("Seconds left:"+ millisUntilFinished / 1000)
            }
            override fun onFinish() {
                textView.setText("Game Over")
            }
        }.start()
        val images: MutableList<Int> = mutableListOf(resizedandroid, resizedcplus, resizedjava,resizedjavascript,
            resizedpython, resizedruby, resizedandroid, resizedcplus, resizedjava,resizedjavascript,
            resizedpython, resizedruby)
        val buttons = arrayOf(button1, button2, button3, button4, button5, button6, button7,
        button8, button9, button10, button11, button12)
        val cardBack = resizedprog
        var clicked=0
        var turnOver = false
        var lastClicked =-1
        images.shuffle()
        for (i in 0..11){
            buttons[i].text="cardBack"
            buttons[i].textSize=0.0F
            buttons[i].setOnClickListener {
                if (buttons[i].text == "cardBack" && !turnOver) {
                    buttons[i].setBackgroundResource(images[i])
                    buttons[i].setText(images[i])
                    if (clicked == 0) {
                        lastClicked = i
                    }
                    clicked++
                }
                else if (buttons[i].text !in "cardBack") {
                    buttons[i].setBackgroundResource(cardBack)
                    buttons[i].setText("cardBack")
                    clicked--
                }

                if (clicked == 2) {
                    turnOver = true
                    if (buttons[i].text == buttons[lastClicked].text) {
                        buttons[i].isClickable = false
                        buttons[lastClicked].isClickable = false
                        turnOver = false
                        clicked = 0
                    }
                } else if (clicked == 0) {
                    turnOver = false
                }
            }
    }

    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        return when (item.itemId) {
            R.id.action_settings -> true
            else -> super.onOptionsItemSelected(item)
        }
    }

    override fun onSupportNavigateUp(): Boolean {
        val navController = findNavController(R.id.nav_host_fragment_content_main)

        return navController.navigateUp(appBarConfiguration)
               || super.onSupportNavigateUp()
   }
}

